users/users/users.php

```diff
@@ -41,158 +36,14 @@ class UsersApiResourceUsers extends ApiResource
-					$eobj->id         = $data['user_id'];
-					$eobj->code       = '200';
-					$eobj->message    = JText::_('COM_USERS_USERS_N_ITEMS_DELETED_1');
-					$this->plugin->setResponse($eobj);
-
-					return;
-			}
-			else
-			{
-					$eobj->code = '400';
-					$eobj->message = JText::_('COM_USERS_USER_NOT_FOUND');
-					$this->plugin->setResponse($eobj);
-
-					return;
-			}
-		}
-		else
-		{
-			// Not given username or user_id to edit the details of user
-			$eobj->code = '400';
-			$eobj->message = JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE');
-			$this->plugin->setResponse($eobj);
-
-			return;
-		}
-	}
-
-	/**
-		* Function for edit user record.
-		*
-		* @return void
-		*/
-	public function put()
-	{
-		$app              = JFactory::getApplication();
-		$db 			  = JFactory::getDbo();
-		$data             = array();
-		$eobj             = new stdClass;
-		$eobj->status     = false;
-		$eobj->id         = 0;
-
-		// Get values paased by put in the url
-		$data = $app->input->getArray(array());
-
-		// Check username or user_id to edit the details of user
-		if (isset($data['username']) || isset($data['user_id']))
-		{
-			if (!$data['user_id'] && isset($data['username']))
-			{
-				// Get user_id with the help of username
-				$query = $db->getQuery(true);
-				$query->select('id');
-				$query->from('#__users');
-				$query->where($db->quoteName('username') . '=' . $db->quote($data['username']));
-				$db->setQuery($query);
-				$user_id = $db->loadResult();
-				$data['user_id'] = $user_id;
-			}
-
-			// Given username or user_id not exist
-				if (!$data['user_id'])
-				{
-					$eobj->code = '400';
-					$eobj->message = JText::_('COM_USERS_USER_NOT_FOUND');
-					$this->plugin->setResponse($eobj);
-
-					return;
-				}
-
-				$user = JFactory::getUser($data['user_id']);
-
-				// Bind the data.
-				if (!$user->bind($data))
-				{
-					// User deatils are not updated
-					$message = $user->getError();
-					$eobj->code = '400';
-					$eobj->message = $message;
-					$this->plugin->setResponse($eobj);
-
-					return;
-				}
-
-				// Save the user data
-				if (!$user->save())
-				{
-					// User deatils are not updated
-					$message = $user->getError();
-					$eobj->code = '400';
-					$eobj->message = $message;
-					$this->plugin->setResponse($eobj);
-
-					return;
-				}
-				else
-				{
-					// Updated records updated successsfully
-					$eobj->status     = true;
-					$eobj->id         = $data['user_id'];
-					$eobj->code       = '200';
-					$eobj->message    = JText::_('PLG_API_USERS_ACCOUNT_EDITED_SUCCESSFULLY_MESSAGE');
-					$this->plugin->setResponse($eobj);
-
-					return;
-				}
-		}
-		else
-		{
-			// Not given username or user_id to edit the details of user
-			$eobj->code = '400';
-			$eobj->message = JText::_('PLG_API_USERS_REQUIRED_DATA_EMPTY_MESSAGE');
-			$this->plugin->setResponse($eobj);
-
-			return;
-		}
+		$this->plugin->setResponse(JText::_('PLG_API_USERS_IN_DELETE_FUNCTION_MESSAGE'));
```

Comment:
@coolbung @Pratikshakhandagale why did we remove delete code.?

Reply:
I believe because it wasn't robust. The plugin triggers weren't firing or something.. dont remember now. But ok if we can verify and bring it back. 

However the bigger problem with this resource is that it's hardwired to work with Easysocial. That needs to be fixed.